samples/bot-sso-adaptivecard/csharp/BotSsoAdaptivecard/Bots/DialogBot.cs

```diff
@@ -0,0 +1,294 @@
+
+                if (actiondata["verb"] == null)
+                    return null;
+
+                string verb = actiondata["verb"].ToString();
+                JObject authentication = null;
+                if (value["authentication"] != null)
+                {
+                    authentication = JsonConvert.DeserializeObject<JObject>(value["authentication"].ToString());
+                    string token = authentication["token"].ToString();
+                    var userTokenClient = turnContext.TurnState.Get<UserTokenClient>();
+                    var tokenResource = await userTokenClient.ExchangeTokenAsync(turnContext.Activity.From.Id, _connectionName, turnContext.Activity.ChannelId, new TokenExchangeRequest(null, token), cancellationToken).ConfigureAwait(false);
+                }
+                string state = null;
+                if (value["state"] != null)
+                {
+                    state = value["state"].ToString();
+                }
+                // authToken and state are absent, handle verb
+                if (authentication == null && state == null)
+                {
+                    switch (verb)
+                    {
+                        case "initiateSSO":
+                            return await initiateSSOAsync(turnContext, cancellationToken);
+                    }
+                }
+                else
+                {
+                    return await createAdaptiveCardInvokeResponseAsync(authentication, state, turnContext,cancellationToken);
+
+                }
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Authentication success.
+        /// AuthToken or state is present. Verify token/state in invoke payload and return AC response
+        /// </summary>
+        /// <param name="authentication"></param>
+        /// <param name="state"></param>
+        /// <param name="turnContext"></param>
+        /// <param name="cancellationToken"></param>
+        /// <param name="isBasicRefresh"></param>
+        /// <param name="fileName"></param>
+        /// <returns></returns>
+        private async Task<InvokeResponse> createAdaptiveCardInvokeResponseAsync(JObject authentication, string state, ITurnContext<IInvokeActivity> turnContext, CancellationToken cancellationToken, bool isBasicRefresh = false, string fileName = "adaptiveCardResponseJson.json")
+        {
+            //verify token is present or not
+
+            bool isTokenPresent = authentication != null ? true : false;
+            bool isStatePresent = state != null && state != "" ? true : false;
+
+            // TODO : Use token or state to perform operation on behalf of user
+
+            string[] filepath = { ".", "Resources", fileName };
+
+            var adaptiveCardJson = File.ReadAllText(Path.Combine(filepath));
+            AdaptiveCardTemplate template = new AdaptiveCardTemplate(adaptiveCardJson);
```

Comment:
remove  lines from 192 to 210. We dont need to call graph and all. We can just send normal adaptive in response here

Reply:
**Execute pipeline**
![image](https://user-images.githubusercontent.com/117716554/212266359-a04d4107-0ea6-4749-8438-9da3d1a5bab2.png)